### PR TITLE
Merge Main arm beta back into Main

### PR DESCRIPTION
**Pull request by jamesi8086**
## Description

There are some changes left on main-arm-beta that should be examined before merging back into main

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Be detailed so that your code reviewer can understand exactly how much and what kinds of testing were done, and which might still be worthwhile to do.

- [x] Unit tests: [Add your description here]
- [x] Simulator testing: [Add your description here]
- [x] On-robot bench testing: [Add your description here]
- [x] On-robot field testing: [Add your description here]

### CODE REVIEW

File: src/main/java/com/team766/controllers/PIDController.java [hunk 1]
```diff
@@ -269,9 +270,63 @@ public void calculate(double cur_input) {
 		pr("	Total Error: " + total_error + "		Current Error: " + cur_error +
 		   "	Output: " + output_value + " 	Setpoint: " + setpoint);
 	}
+	public void calculate(double cur_input, double target_speed) {
```
Comment (jamesi8086) [inline src/main/java/com/team766/controllers/PIDController.java:273]: unused, ignore

File: build.gradle [hunk 2]
```diff
@@ -62,7 +62,7 @@ deploy {
                     //    # There is insufficient memory for the Java Runtime Environment to continue.
                     //    # Native memory allocation (mmap) failed to map 1048576 bytes for committing
                     //    reserved memory.
-                    jvmArgs.add("-XX:MinRAMPercentage=15")
+                    jvmArgs.add("-XX:MinRAMPercentage=10")
```
Comment (jamesi8086) [inline build.gradle]: needs to be examined @rcahoon

File: src/main/java/com/team766/robot/mechanisms/ArmsAnimation.java [hunk 3]
```diff
@@ -0,0 +1,67 @@
+package com.team766.robot.mechanisms;
```
Comment (jamesi8086) [inline src/main/java/com/team766/robot/mechanisms/ArmsAnimation.java:1]: unused, ignore

File: src/main/java/com/team766/robot/mechanisms/Grabber.java [hunk 4]
```diff
@@ -3,23 +3,28 @@
...
-	private MotorController wrist;
-	private SolenoidController grabber;
+public class Grabber extends Mechanism {
+	private MotorController grabby;
 
 	public Grabber(){
-		wrist = RobotProvider.instance.getMotor("wrist");
-		grabber = RobotProvider.instance.getSolenoid("grabber");
+		grabby = RobotProvider.instance.getMotor("arms.grabber");
+		grabby.setCurrentLimit(10.0);
```
Comment (jamesi8086) [inline src/main/java/com/team766/robot/mechanisms/Grabber.java:12]: @1yd1a could be reduced to 5-8 amps to reduce mechanism wear during stall

**Review by maxspier — APPROVED** [review]

**Review by rajitzg — APPROVED** [review]
Reviewed with max in person

**Pull request merged**